src/commands/mentor/mentores.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  CommandInteraction,
+  SlashCommandBuilder,
+  TextChannel,
+  AutocompleteInteraction,
+  roleMention,
+  channelMention,
+} from 'discord.js'
+
+export const data = new SlashCommandBuilder()
+  .setName('mentores')
+  .setDescription('Pedir ajuda dos mentores')
+  .addStringOption((opt) =>
+    opt
+      .setName('mentor-role')
+      .setDescription('Escolha o tipo de mentor que quer tirar sua dúvida')
+      .setAutocomplete(true)
+      .setRequired(true)
+  )
+  .addStringOption((opt) =>
+    opt
+      .setName('pergunta')
+      .setDescription('Digite a sua dúvida')
+      .setRequired(true)
+  )
+
+export async function autocomplete(interaction: AutocompleteInteraction) {
+  const focused = interaction.options.getFocused()
+
+  const mentorRolesArray =
+    interaction.guild?.roles.cache.map((r) => ({
+      name: r.name,
+      value: r.id,
+    })) || []
+  const mentorRoles = mentorRolesArray.filter((r) =>
+    r.name.toLowerCase().includes('mentor')
+  )
+  const mappedMentorRoles =
+    mentorRoles.map((r) => ({
+      name: r.name,
+      id: r.value,
+    })) || []
+
+  const filtered = mappedMentorRoles.filter((choice) =>
+    choice.id.toLowerCase().includes(focused.toLowerCase())
+  )
+
+  await interaction.respond(
+    filtered.map((choice) => ({
+      name: choice.name,
+      value: choice.id,
+    }))
+  )
+}
+
+export async function execute(interaction: CommandInteraction) {
+  const channel = interaction.channel as TextChannel
```

Comment:
acho que a gnt tem que verificar se o canal é instance of textchannel

se forçar com `as` e passar por outro canal sem ser text, quebra o bot

Reply:
Mas tipo na linha 71 eu verifico se o canal em que ta sendo chamado o comando é o help então mesmo que a pessoa tente usar em outro de voz ela não consegue. Agora escrevendo isso minha dúvida é: e se tivermos um canal de voz com o mesmo nome do canal de texto help?

Reply:
Adcionei a verificação se é um canal de texto.